src/suffer.cpp

```diff
@@ -601,129 +624,130 @@ void Character::suffer_from_schizophrenia()
     }
 }
 
-void Character::suffer_from_asthma( const int current_stim )
+void suffer::from_asthma( Character &you, const int current_stim )
 {
-    if( has_effect( effect_adrenaline ) ||
-        has_effect( effect_datura ) ||
-        has_effect( effect_took_antiasthmatic ) ) {
+    if( you.has_effect( effect_adrenaline ) ||
+        you.has_effect( effect_datura ) ||
+        you.has_effect( effect_took_antiasthmatic ) ) {
         return;
     }
     //cap asthma attacks to 1 per minute (or risk instantly killing players that rely on oxygen tanks)
     if( !one_in( std::max( to_turns<int>( 1_minutes ),
                            ( to_turns<int>( 6_hours ) - current_stim * 300 ) ) *
-                 ( has_effect( effect_sleep ) ? 10 : 1 ) ) ) {
+                 ( you.has_effect( effect_sleep ) ? 10 : 1 ) ) ) {
         return;
     }
-    bool auto_use = has_charges( itype_inhaler, 1 ) || has_charges( itype_oxygen_tank, 1 ) ||
-                    has_charges( itype_smoxygen_tank, 1 );
-    bool oxygenator = has_bionic( bio_gills ) && get_power_level() >= ( bio_gills->power_trigger / 8 );
-    if( underwater ) {
-        oxygen = oxygen / 2;
+    bool auto_use = you.has_charges( itype_inhaler, 1 ) || you.has_charges( itype_oxygen_tank, 1 ) ||
+                    you.has_charges( itype_smoxygen_tank, 1 );
+    bool oxygenator = you.has_bionic( bio_gills ) &&
+                      you.get_power_level() >= ( bio_gills->power_trigger / 8 );
+    if( you.underwater ) {
+        you.oxygen = you.oxygen / 2;
         auto_use = false;
     }
 
-    add_msg_player_or_npc( m_bad, _( "You have an asthma attack!" ),
-                           "<npcname> starts wheezing and coughing." );
+    you.add_msg_player_or_npc( m_bad, _( "You have an asthma attack!" ),
+                               "<npcname> starts wheezing and coughing." );
 
     map &here = get_map();
-    if( in_sleep_state() && !has_effect( effect_narcosis ) ) {
+    if( you.in_sleep_state() && !you.has_effect( effect_narcosis ) ) {
         inventory map_inv;
-        Character &player_character = get_player_character();
-        map_inv.form_from_map( player_character.pos(), 2, &player_character );
+        map_inv.form_from_map( you.pos(), 2, &you );
```

Comment:
The sudden jump to using `player_character` throughout this function looked suspect to me, assuming this function is intended to work with any Character - so I replaced the references with the same `you` instance used by the rest of the function. It makes sense that we should be looking nearby the Character who needs to treat their asthma, rather than nearby the player character who may be in a different location.

Please let me know if there's some reason I should not have done this.